src/model.ts

```diff
@@ -29,14 +44,21 @@ export type MarkdownNil = false | null | undefined;
 /** Any primitive Markdown type. */
 export type MarkdownNode = MarkdownNil | MarkdownText | MarkdownElement;
 
+/** Promise Markdown type that resolves to nested Markdown type. */
+export type MarkdownPromise = Promise<MarkdownChildren>;
+
 /** Nested Markdown type. */
-export type MarkdownChildren = MarkdownNode | MarkdownChildren[];
+export type MarkdownChildren =
+  | MarkdownNode
+  | MarkdownChildren[]
+  | MarkdownPromise;
 
 /** Helper type for creating Elements that accept other markdown as children. */
 export type PropsWithChildren<
-  AdditionalProps extends unknown = unknown
+  AdditionalProps extends unknown = unknown,
+  Children = MarkdownChildren
 > = AdditionalProps & {
-  children?: MarkdownChildren;
+  children?: Children;
```

Comment:
This change should not be necessary as MarkdownChildren already contains MarkdownPromise